#############################################
#   Mapping                                 #
#############################################
astar:
  algorithm_type: "astar" 
  visualize_process_en: true
  map_voxel_num: [600,600,100] 
  allow_diag: true 
  heu_type: 2 
  example_start: [-10.0, -5.0, 0.7]
  example_goal: [10.0, 5.0, 0.7]
  # 添加标定点列表和到达阈值
  waypoints:
    - [2.77931, -0.576113, 0.7]  # 第一组起点
    - [31.803, -2.52213, 0.7]  # 第一组终点
    - [28.5576, -4.55573, 0.7]  # 第二组起点
    - [1.98401, -2.8763, 0.7]  # 第二组终点
    - [2.97804, -5.25965, 0.7]  # 第三组起点
    - [31.4704, -7.35594, 0.7]  # 第三组终点
  arrival_threshold: 1.0  # 到达阈值（米）
rog_map:
  resolution: 0.2                    # 增大分辨率降低计算负载
  inflation_resolution: 0.2         # 缩小膨胀范围
  inflation_step: 3
  unk_inflation_en: false 
  unk_inflation_step: 1
  map_size: [60,60,10] 
  fix_map_origin: [0,0,0] 
  frontier_extraction_en: true

  virtual_ceil_height: 9999 
  virtual_ground_height: -9999 

  load_pcd_en: false 
  map_sliding: 
    enable: false
    threshold: 0.3

  esdf:
    enable: true
    resolution: 0.3               # 确保 esdf.resolution ≥ rog_map.resolution
    local_update_box: [30,30,5] 

  ros_callback:
    enable: true
    cloud_topic: "/cloud_registered"
    odom_topic: "/Odometry"
    odom_timeout: 3.0

  visualization:
    enable: true
    use_dynamic_reconfigure: false
    time_rate: 10
    frame_rate: 0
    range: [100,100,10] 
    frame_id: "world"
    pub_unknown_map_en: false

  intensity_thresh: -10
  point_filt_num: 10                 # 增大降采样率

  raycasting:
    enable: true
    batch_update_size: 1
    local_update_box: [60,60,10] 
    ray_range: [0.0, 5] 
    p_min: 0.12
    p_miss: 0.49
    p_free: 0.499
    p_occ: 0.85
    p_hit: 0.9
    p_max: 0.98
    unk_thresh: 1.0
